projects/solace-finance/index.js

```diff
@@ -35,36 +39,52 @@ const ETHEREUM_TOKENS = [
   },
 ];
 
-async function ethereum(timestamp, block, chainBlocks) {
+async function fetchBalances(contract, tokens, chain, block) {
   const balances = {};
+
   const { output: _tvlList } = await sdk.api.abi.multiCall({
-    calls: ETHEREUM_TOKENS.map((token) => ({
+    calls: tokens.map((token) => ({
       target: token.PoolToken,
-      params: [ETHEREUM_UWP_ADDRESS],
+      params: [contract],
     })),
     abi: "erc20:balanceOf",
-    block: chainBlocks["ethereum"],
-    chain: "ethereum",
+    block: block,
+    chain: chain,
   });
 
   _.each(_tvlList, async (element) => {
     let address = element.input.target;
     let balance = element.output;
 
-    if (BigNumber(balance).toNumber() <= 0) {
-      return;
+    if (chain === "polygon" || chain === "aurora") {
+      if (address === "0xDA2585430fEf327aD8ee44Af8F1f989a2A91A3d2") {
+        // Frax token on aurora network is not working with adapter.
+        // So, Frax balance is added to DAI.
+        address = "0xe3520349F477A5F6EB06107066048508498A291b";
+      }
+      address = chain + ":" + address;
     }
-
-    balances[address] = BigNumber(balances[address] || 0)
-      .plus(balance)
-      .toFixed();
+    
+    sdk.util.sumSingleBalance(balances, address, balance);
   });
   return balances;
 }
 
+async function ethereum(timestamp, block, chainBlocks) {
+  let result = await fetchBalances(ETHEREUM_UWP_ADDRESS, ETHEREUM_TOKENS,  "ethereum", chainBlocks["ethereum"]);
+  const lpBalances = await fetchBalances(SOLACE_USDC_POOL, ETHEREUM_TOKENS,  "ethereum", chainBlocks["ethereum"]);
+  result[ETHEREUM_LP_TOKENS.SOLACE] = BigNumber(result[ETHEREUM_LP_TOKENS.SOLACE] || 0).plus(lpBalances[ETHEREUM_LP_TOKENS.SOLACE] || 0).toFixed();
```

Comment:
do you need to unwrap these LP tokens using unwrapUniswapLPs()?

Reply:
No we don't need actually. I tried it but it didn't work. We're on multiple chains and using different lp structures.